internal/resources/cloudprovider/resource_aws_cloudwatch_scrape_job.go

```diff
@@ -309,32 +309,26 @@ func (r *resourceAWSCloudWatchScrapeJob) Read(ctx context.Context, req resource.
 }
 
 func (r *resourceAWSCloudWatchScrapeJob) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	var stateData awsCWScrapeJobTFModel
-	diags := req.State.Get(ctx, &stateData)
-	resp.Diagnostics.Append(diags...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	var configData awsCWScrapeJobTFModel
-	diags = req.Config.Get(ctx, &configData)
+	var planData awsCWScrapeJobTFModel
+	diags := req.Plan.Get(ctx, &planData)
```

Comment:
so to understand, there's three things here:
- req.State => data in the tf state (eg last applied)
- req.Config => what the hcl code describes right now at the moment of plan / apply
- req.Plan => mixin of the above, and what terraform show's as planned / intends to apply.

In another words, state is the before, config is the desired, and "`plan = state + config`"?

Reply:
mostly, `plan = state + config`, with any attributes with the `Computed` schema modifier not filled in by those two determined to be "(known after apply)"